OpenCore 1.0.4 + Sonoma BlueTooth fixed + support OTA

Shuery-Shuai opened the pull request:
- Upgrade OpenCore and kexts to the latest releases.
- Support OTA or installation to macOS Sonoma.
- Add a note for macOS Sonoma installation/update.
- Update kexts info in the README file. Also removed the release date of them (It's too much work to maintain it, but of course I can add it back if you want to keep it).

Shuery-Shuai committed:
update README.md with line reference for SecureBootModel setting

Shuery-Shuai committed:
kexts: update BlueToolFixup to 2.7.0

Shuery-Shuai committed:
remove useless file `.DS_Store`

Shuery-Shuai committed:
OC: Update to 1.0.4

Shuery-Shuai committed:
kexts: update AppleALC to 1.9.4

Shuery-Shuai committed:
kexts: change BlueToolFixup to @zxystd 's version.
refer: https://github.com/OpenIntelWireless/IntelBluetoothFirmware/issues/437#issuecomment-1579931908

Shuery-Shuai committed:
kexts: update ECEnabler to 1.0.5

Shuery-Shuai committed:
kexts: update IntelBluetoothFirmware to 2.4.0

Shuery-Shuai committed:
kexts: update IntelBTPatcher to 2.4.0

Shuery-Shuai committed:
kexts: update Lilu to 1.7.0

Shuery-Shuai committed:
kexts: update NVMEFix to 1.1.2

Shuery-Shuai committed:
kexts: update RealtekRTL8111 to 2.5.0

Shuery-Shuai committed:
kexts: add RestrictEvents 1.1.5

Shuery-Shuai committed:
kexts: update SMCBatteryManager to 1.3.6

Shuery-Shuai committed:
kexts: update SMCProcessor to 1.3.6

Shuery-Shuai committed:
exts: update SMCSuperIO to 1.3.6

Shuery-Shuai committed:
kexts: update USBToolBox to 1.1.1

Shuery-Shuai committed:
kexts: update VirtualSMC to 1.3.6

Shuery-Shuai committed:
kexts: update VoodooPS2Controller to 2.3.7

Shuery-Shuai committed:
kexts: update VoodooRMI to 1.4.2

Shuery-Shuai committed:
kexts: update VoodooSMBus to 3.0

Shuery-Shuai committed:
kexts: update WhateverGreen to 1.6.9

Shuery-Shuai committed:
config: add NVRAM boot-args `revpatch=sbvmm`

Shuery-Shuai committed:
update README.md

- Add notes of install/update Sonoma.
- Remove the data in Details of kexts 'cause it is too complicated to maintain.
- Add kexts info: IntelBTPatcher, RestrictEvents.
- Update kexts info: RRTL8111 named to RealtekRTL8111.

Shuery-Shuai committed:
update README.md

ananjaser1211 commented:
This looks very nice, i am hopeful ill get a chance to give it a spin on my HP, Thank you for the contribution, will see if @TECHNIKVERBOT or @SkyrilHD have any additional input, otherwise will merge soon.

SkyrilHD commented on README.md:
```diff
@@ -41,40 +50,45 @@
 ## OpenCore
-- OpenCore v0.9.5 (Release - 09/11/2023)
+
+- OpenCore v0.8.8 (Release - 01/02/2023)
```
Are we going back in time?

SkyrilHD commented on README.md:
```diff
@@ -2,24 +2,33 @@
+> 1. **Before Installation**: Set [`Misc` -> `Security` -> `SecureBootModel`](/EFI/OC/config.plist#L1028) to `Disabled`.
+> 2. **After Successfully Booting**: Change `SecureBootModel` back to `j132`.
+>
+>    ⚠️ Only change back to `j132` after **successfully reaching the desktop**, otherwise boot may fail!
```
Any reason why we have to use `j132` specifically instead of `Default`?

SkyrilHD commented on README.md:
```diff
@@ -2,24 +2,33 @@
+> [!IMPORTANT]  
+> For macOS **Sonoma** Installation/Update:
+>
+> 1. **Before Installation**: Set [`Misc` -> `Security` -> `SecureBootModel`](/EFI/OC/config.plist#L1028) to `Disabled`.
```
L1028 could be changed if the config is altered (e.g. if a new property is added by an OpenCore update). Use a permalink instead.

SkyrilHD commented on EFI/OC/config.plist:
Can you fix the indention of this file?

Few things that stood out:
- You changed the kext orders. I am going to assume it works.
- You added kext versions inside the comments. This makes it harder to maintain. If possible, can you remove it? Either leave it empty or use kext file names (as used in PorperTree). I prefer the first option.
- You can move the revpatch=sbvmm boot-arg as a NVRAM variable. (That's how I did it: [Link](https://github.com/SkyrilHD/Dell-E7250-Hackintosh/blob/15.0/OpenCore/EFI/OC/config.plist#L1340-L1341))
- You do not need to add NVRAM variables for Bluetooth since BlueToolFixup 2.6.8.
- Do not leave `UEFI>Output>InitialMode` empty. Add `Auto` as value.

SkyrilHD reviewed (changes requested):
Few changes are needed. Check the comments.

Shuery-Shuai committed:
config: change SecureBootModel to Default

Big Sur and Above (11.0+): The recommended value is Default: https://dortania.github.io/OpenCore-Install-Guide/config.plist/security.html#security-securebootmodel

Shuery-Shuai committed:
config: set InitialMode to Auto

Shuery-Shuai committed:
config: move revpatch=sbvmm as a NVRAM variable

Shuery-Shuai committed:
config: remove kexts version in comment

Shuery-Shuai committed:
config: remove NVRAM variables for BluetoothFixup

Shuery-Shuai committed:
update README.md

- update the OpenCore version and delete the release time.
- change SecureBootModel from j132 to Default.
- delete the link to config.plist.

Shuery-Shuai commented:
Sorry for taking so long to reply. Here are the changes submitted this time:

1. **OpenCore version**: Update the OpenCore version in `README.md` and delete the release time.
2. **`SecureBootModle`**: When I was looking for how to upgrade macos from Ventura to Sonoma, I found this solution and tried it, and was able to successfully enter the system. Now it is tested that changing `j132` to `Default` can also enter the system, and the [official document](https://dortania.github.io/OpenCore-Install-Guide/config.plist/security.html#security-securebootmodel) also recommends Big Sur and above systems to use `Default`, so this commit changes it back to `Default`.
3. **Link to `config.plist` in README**: The link has been deleted in this commit.
4. **Kext version information in `Comment`**: Deleted.
5. **`revpatch=sbvmm` in `boot-args`**: Moved to `NVRAM` variable.
6. **`NVRAM` variables for `BluetoothFixup`**: Deleted.
7. **`InitialMode`**: set to `Auto`.

SkyrilHD reviewed (approved):
LGTM! Thank you very much for this PR.
